### PR TITLE
First version of setup script

### PR DESCRIPTION
**Pull request by enolredhat**

### CODE REVIEW

File: setup/setup.sh [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    fi
+
+    cd $CERT_DIRECTORY
+    openssl genrsa -out $USERNAME.key 2048 &> /dev/null
+    openssl req -new -key $USERNAME.key -out $USERNAME.csr -subj "/CN=$USERNAME/O=group1" &> /dev/null
+    openssl x509 -req -in $USERNAME.csr -CA $MINIKUBE_DIR/ca.crt -CAkey $MINIKUBE_DIR/ca.key -CAcreateserial -out $USERNAME.crt -days 500 &> /dev/null
+}
+
+function create_namespace() {
+    echo "Creating namespace 'redhat-test'..."
```
Comment (someth2say) [inline setup/setup.sh:19]: This namespace looks weird.
To align with Developer Sandbox we need to create something like `username-dev` and `username-stage`.

**Comment by someth2say** [comment]
Maybe we can also make a single file for both manifests?

File: setup/setup.sh (hunk 1 above)
Comment (flozanorht) [reply setup/setup.sh:19]: +1

File: setup/files/role.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  namespace: redhat-test
+  name: pod-reader
+rules:
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "watch", "list"]
```
Comment (flozanorht) [inline setup/files/role.yml:9]: That role looks too restrictice. We want something that allows a developer user to deploy apps, so create and edit pods, services, etc. I thought that the OpenShift read, edit, and admin roles were standard from Kubernetes RBAC. If not, we can use the edit role as a starting point and simplify it a bit by removing references to openshift extensions.

File: setup/setup.sh [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+if ! kubectl config use-context $USERNAME-context &> /dev/null ; then
+    echo "New context is not available"
+    exit
+fi
+
+cp $USERNAME.key $MINIKUBE_DIR/$USERNAME.key
+cp $USERNAME.crt $MINIKUBE_DIR/$USERNAME.crt
+
+echo "OK!"
```
Comment (flozanorht) [inline setup/setup.sh:88]: setup.sh makes the assumption that it runs under bash and it uses bash extensions. As our audience if not constrained to a workstation VM that we control, I would be extra careful and check it is running under bash, the same as you check that openssl and other commands are available. And kudos for doing defensive programming!

Is there a way of checking that minikube is actually installed in the directory, and of checking that minikube is up before trying to set up contexts and create resources?

A nice final step would be adding a verification that the new user do work as intended, and print instructions for using it and for swithching to kubeadmin. Maybe that belongs to the lecture, I just like to be extra helpful with beginners.

**Comment by someth2say** [comment]
Superseded by https://github.com/RedHatTraining/DO100-apps/pull/9

**Pull request closed without merging**